features/upgrade/storage/upgrade.feature

```diff
@@ -344,3 +344,222 @@ Feature: Storage upgrade tests
+    Then the step should succeed
+    And I execute on the pod:
+      | sh | -c | sync -f /mnt/csipath/test-before-upgrade |
+    
+    Given I obtain test data file "storage/csi/volumesnapshotclass-aws.yaml"
+    When I run oc create over "volumesnapshotclass-aws.yaml" replacing paths:
+      | ["metadata"]["name"] | csi-aws-ebs-snapshotclass-prepare |
+    Then the step should succeed
+   
+    Given I obtain test data file "storage/csi/volumesnapshot_v1.yaml"
+    When I run oc create over "volumesnapshot_v1.yaml" replacing paths:  
+      | ["metadata"]["name"]                            | csi-aws-ebs-snapshot-prepare      |
+      | ["spec"]["volumeSnapshotClassName"]             | csi-aws-ebs-snapshotclass-prepare |      
+      | ["spec"]["source"]["persistentVolumeClaimName"] | mypvc-ori-beforeupgrade-prepare   |
+    Then the step should succeed
+    And I wait up to 200 seconds for the steps to pass:
+   
+    """
+    When I run the :describe client command with:
+      | resource | volumesnapshot               |
+      | name     | csi-aws-ebs-snapshot-prepare |
+    Then the output should match "Ready To Use\:\s+true"
+    """
+    
+    Given I obtain test data file "storage/csi/restorepvc.yaml"
+    Then I run oc create over "restorepvc.yaml" replacing paths:
+      | ["metadata"]["name"]           | mypvc-res-beforeupgrade-prepare |
+      | ["spec"]["storageClassName"]   | gp2-csi                         |
+      | ["spec"]["dataSource"]["name"] | csi-aws-ebs-snapshot-prepare    |
+    Then the step should succeed
+
+    Given I obtain test data file "storage/misc/deployment.yaml"
+    When I run oc create over "deployment.yaml" replacing paths:
+      | ["metadata"]["name"]                                                             | mydep-res-beforeupgrade-prepare |
+      | ["spec"]["template"]["metadata"]["labels"]["action"]                             | mydep-res-beforeupgrade-prepare |
+      | ["spec"]["template"]["spec"]["volumes"][0]["persistentVolumeClaim"]["claimName"] | mypvc-res-beforeupgrade-prepare |
+      | ["spec"]["template"]["spec"]["containers"][0]["volumeMounts"][0]["mountPath"]    | /mnt/csipath                    | 
+    
+    Then the step should succeed
+    And a pod becomes ready with labels:
+      | action=mydep-res-beforeupgrade-prepare |     
+ 
+    When I execute on the pod:
+      | ls | /mnt/csipath |
+    Then the output should contain:
+      | test-before-upgrade |
+
+  # @author ropatil@redhat.com
+  # Description:
+  # Check csi snapshot controller status after upgrade
+  # Use project, create restore pvc/deployment, file(test-after-upgrade) in deployment
+  # Check both files(test-before-upgrade,test-after-upgrade) in dep 
+  # Create new project, new pvc/dep, file(test-after-upgrade) in dep
+  # Create new Volumesnapshot Class, Volumesnapshot
+  # Create new Restore pvc/dep,file(test-after-upgrade-check) in dep
+  # Check files(test-after-upgrade,test-after-upgrade-check) in dep
```

Comment:
Do you mind removing these comments from the scenario header? Comments could be added to right before each step.